tools/olm-bundle/generate-olm-bundle.sh

```diff
@@ -0,0 +1,120 @@
+
+  # Update namespaced RBAC rules with Cluster Operator roles + Entity Operator roles
+  yq ea -i 'select(fi==0).spec.install.spec.permissions[0].rules = select(fi==1).rules + select(fi==2).rules | select(fi==0)' \
+	  ${CSV_FILE} \
+	  ${CRD_DIR}/020-ClusterRole-strimzi-cluster-operator-role.yaml \
+	  ${CRD_DIR}/031-ClusterRole-strimzi-entity-operator.yaml
+   
+  yq ea -i ".spec.replaces = \"${PREVIOUS_BUNDLE_VERSION}\" | .spec.replaces style=\"\"" ${CSV_FILE}
+  
+  # Copy Strimzi client roles
+  cp ${CRD_DIR}/033-ClusterRole-strimzi-kafka-client.yaml ${MANIFESTS}/strimzi-kafka-client_rbac.authorization.k8s.io_v1_clusterrole.yaml
+  
+  generate_related_images
+  generate_image_digests
+
+  rm bundle.Dockerfile
+  rm ${MANIFESTS}/strimzi-cluster-operator_v1_serviceaccount.yaml
+}
+
+validate_olm_bundle() {
+  operator-sdk bundle validate ./bundle/ --select-optional name=operatorhub
+}
+
+generate_related_images() {
+  yq ea -i '.spec.relatedImages = null' ${CSV_FILE}
+
+  ENV="*-operator *BRIDGE_IMAGE *JMXTRANS_IMAGE *EXECUTOR_IMAGE"
+  for env in $ENV; 
+  do
+    image=$(yq ea ".. | select(has(\"name\")).env[] | (select (.name == \"$env\")).value" ${CSV_FILE})
+    if [ -z "$image" ]; then
+      # Find operator image
+      image=$(yq ea ".. | select(has(\"image\")) | (select (.name == \"$env\")).image" ${CSV_FILE} | head -n 1) 
+      # Set operator image in annotations
+      yq ea -i ".metadata.annotations.containerImage = \"$image\"" ${CSV_FILE}
+    fi
+    name="strimzi-$(echo $image | cut -d':' -f1 | cut -d'@' -f1 | rev | cut -d'/' -f1 | rev)";
+    yq ea -i ".spec.relatedImages += [{\"name\": \"$name\", \"image\": \"$image\"}]" ${CSV_FILE};
+  done
+
+  # Add Kafka images to relatedImages section.
+  KAFKA_IMAGE_VALUES=$(yq eval '.. | select(has("name")).env[] | (select (.name == "STRIMZI_KAFKA_IMAGES")).value' ${CSV_FILE})
+  for val in $KAFKA_IMAGE_VALUES;
+  do
+    name="strimzi-kafka-$(echo "$val" | cut -d'=' -f1 | tr -d '.')";
+    image=$(echo "${val}" | cut -d'=' -f2);
+    yq ea -i ".spec.relatedImages += [{\"name\": \"$name\", \"image\": \"$image\"}]" ${CSV_FILE};
+  done
+}
+
+# Iterates through images listed in newly generated relatedImages section
+# and replaces image tags with image digests
+generate_image_digests() {
+
+  IMAGES=$(yq eval '.spec.relatedImages[].image' "${CSV_FILE}")
+  for image_tag in $IMAGES;
+  do
+    tag=$(echo $image_tag | cut -d':' -f2);
+    image=$(echo $image_tag | cut -d':' -f1);
+    
+    registry=$(echo $image | cut -d'/' -f1);
+    org=$(echo $image | rev | cut -d'/' -f2 | rev);
+    repo=$(echo $image | rev | cut -d'/' -f1 | rev);
+    
+    echo "Get digest from remote registry for: $image_tag"; 
+    digest="$(curl -s -H "Accept: application/vnd.docker.distribution.manifest.v2+json" https://${registry}/v2/${org}/${repo}/manifests/${tag} | sha256sum | head -c 64)";
+    
+    image_digest="${image}@sha256:${digest}"
+    sed -i "s|${image_tag}|${image_digest}|g" ${CSV_FILE};
```

Comment:
You should probably load the multiplatform support and use `$SED` and `$CP` vars to make sure this works on MacOS.

Reply:
https://gist.github.com/see-quick/57aa46cfb46e5598dbec83f7819299d6 something like this should be great

Reply:
You mean like this? https://github.com/strimzi/strimzi-kafka-operator/blob/main/tools/multi-platform-support.sh :-o